frontend/src/components/IssueV1/components/HeaderSection/Assignee/Assignee.vue

```diff
@@ -134,11 +143,145 @@ const changeAssigneeUID = async (uid: string | undefined) => {
 };
 
 const filterAssignee = (user: User): boolean => {
-  return (
-    assigneeCandidates.value.findIndex(
-      (candidate) => candidate.name === user.name
-    ) >= 0
+  return isMemberOfProjectV1(issue.value.projectEntity.iamPolicy, user);
+};
+
+const mapUserOptions = (users: User[]) => {
+  const project = issue.value.projectEntity;
+
+  // Project owners go top
+  users = orderBy(
+    users,
+    [(user) => (isOwnerOfProjectV1(project.iamPolicy, user) ? -1 : 1)],
+    ["asc"]
+  );
+
+  const phase = isCreating.value
+    ? "PREVIEW"
+    : reviewContext.done.value
+    ? "CD"
+    : "CI";
+  const groups: SelectGroupOption[] = [];
+
+  const mapUserOption = (user: User) => ({
+    user,
+    value: extractUserUID(user.name),
+    label: user.title,
+  });
+
+  const members = users.filter((user) =>
+    isMemberOfProjectV1(project.iamPolicy, user)
+  );
+
+  if (phase === "PREVIEW") {
+    const owners = members.filter((user) =>
```

Comment:
Sadly, this might not offer the optimal experience, but it's the best we can achieve.